internal/cmd/operator-sdk/generate/internal/relatedimages.go

```diff
@@ -125,13 +137,15 @@ func (c *relatedImageCollector) collect(name, imageRef, containerRef string) {
 
 func (c *relatedImageCollector) collectedRelatedImages() []operatorsv1alpha1.RelatedImage {
 	final := make([]operatorsv1alpha1.RelatedImage, 0, len(c.relatedImages))
-
+	namesAddedOnce := make(map[string]bool)
 	for _, relatedImage := range c.relatedImages {
 		name := relatedImage.name
-
 		// Prefix the name with the containerRef on name collisions.
 		if len(c.relatedImagesByName[relatedImage.name]) > 1 {
-			name = relatedImage.containerRef + "-" + name
+			if _, added := namesAddedOnce[name]; added {
```

Comment:
If a related image with the same name was discovered, prefix the instances after first discovered name with its container name, leave the base CSV definition as is (if they don't have more than one relatedImage with the same name) and add it to the final CSV.

Reply:
First discovered name will be coming from baseCSV first.